source/include/azure_iot_json_reader.h

```diff
@@ -0,0 +1,182 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file azure_iot_json_reader.h
+ *
+ * @brief The JSON reader used by the middleware for PnP properties.
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ *
+ */
+
+#ifndef AZURE_IOT_JSON_READER_H
+#define AZURE_IOT_JSON_READER_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "azure_iot_hub_client.h"
+
+#include "azure/core/az_json.h"
+
+/**
+ * Defines symbols for the various kinds of JSON tokens that make up any JSON text.
+ */
+typedef enum AzureIoTJSONTokenType
+{
+    eAzureIoTJSONTokenNONE = AZ_JSON_TOKEN_NONE,
+    eAzureIoTJSONTokenBEGIN_OBJECT = AZ_JSON_TOKEN_BEGIN_OBJECT,
+    eAzureIoTJSONTokenEND_OBJECT = AZ_JSON_TOKEN_END_OBJECT,
+    eAzureIoTJSONTokenBEGIN_ARRAY = AZ_JSON_TOKEN_BEGIN_ARRAY,
+    eAzureIoTJSONTokenEND_ARRAY = AZ_JSON_TOKEN_END_ARRAY,
+    eAzureIoTJSONTokenPROPERTY_NAME = AZ_JSON_TOKEN_PROPERTY_NAME,
+    eAzureIoTJSONTokenSTRING = AZ_JSON_TOKEN_STRING,
+    eAzureIoTJSONTokenNUMBER = AZ_JSON_TOKEN_NUMBER,
+    eAzureIoTJSONTokenTRUE = AZ_JSON_TOKEN_TRUE,
+    eAzureIoTJSONTokenFALSE = AZ_JSON_TOKEN_FALSE,
+    eAzureIoTJSONTokenNULL = AZ_JSON_TOKEN_NULL
+} AzureIoTJSONTokenType_t;
+
+typedef struct AzureIoTJSONReader
+{
+    az_json_reader xCoreReader;
```

Comment:
Should we have internal anonymous struct, similar to Writer?

Reply:
yea I had a reason thinking we needed it to be public but I never fixed. updating